python/ray/train/callbacks/profile.py

```diff
@@ -0,0 +1,76 @@
+import logging
+import tempfile
+from pathlib import Path
+from typing import List, Dict, Optional, Union
+
+from torch.profiler import profile
+
+from ray import train
+from ray.train import TrainingCallback
+from ray.train.callbacks.logging import TrainCallbackLogdirManager
+from ray.train.callbacks.results_preprocessors import IndexedResultsPreprocessor
+from ray.train.constants import PROFILER_KEY
+
+logger = logging.getLogger(__name__)
+
+WORKER_TRACE_DIR_NAME = "pytorch_profiler_worker_traces"
+DRIVER_TRACE_DIR_NAME = "pytorch_profiler"
+
+
+class TorchWorkerProfiler():
+    def __init__(self, trace_dir: Optional[str] = None):
+        default_dir = Path(
+            tempfile.gettempdir()).joinpath(WORKER_TRACE_DIR_NAME)
+        trace_dir = trace_dir or default_dir
```

Comment:
```suggestion
        trace_dir = trace_dir or Path(
            tempfile.gettempdir()).joinpath(WORKER_TRACE_DIR_NAME)
```
This is an incredibly small nit, but we can avoid creating the path if not necessary